tests/compiler/test_typing.py

```diff
@@ -380,3 +380,241 @@ def test_2243_override_optional(snippetcompiler) -> None:
 Child()
         """,
     )
+
+
+def test_deprecate_number(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+entity Test:
+    number val
+end
+        """,
+    )
+    with pytest.warns(
+        TypeDeprecationWarning,
+        match=re.escape("Type 'number' is deprecated, use 'float' or 'int' instead"),
+    ):
+        (_, scopes) = compiler.do_compile()
+
+
+def test_number_type(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+entity Test:
+    number i = 0
```

Comment:
No need to have explicit tests for number I think. Was this meant to be a test for `float`?

Reply:
no there is already one for float, i'll remove this one.